daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/SequenceChild.scala

```diff
@@ -123,7 +123,7 @@ abstract class SequenceChild(protected val sq: SequenceTermBase, child: Term, gr
     import SeparatorSuppressionPolicy._
     import SeparatedSequenceChildBehavior._
     child match {
-      case m: ModelGroup => {
+      case ch @  (_: ModelGroup | _: ElementBase) if ch.isInstanceOf[ModelGroup] || ch.isScalar => {
```

Comment:
I personally find this case more difficult to parse out. Maybe move the content of these two cases in a lazy val or def and referece it in the cases?